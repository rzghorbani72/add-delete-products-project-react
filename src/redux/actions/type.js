export const GET_ERRORS = 'GET_ERRORS';
export const RESET_POSTS = 'RESET_POSTS';
export const UPDATE_POSTS = 'UPDATE_POSTS';
export const GET_POSTS = 'GET_POSTS';
