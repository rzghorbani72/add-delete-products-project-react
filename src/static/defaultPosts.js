module.exports = {
	get: () => {
		return Promise.resolve(
			[{
				id:1,
				image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D9%82%D9%86%D8%A7%D8%AF%DB%8C-%D9%86%D8%A7%D8%AA%D9%84%DB%8C-600x399.jpg',
				title:'قنادی ناتلی',
				averagePrice:'24000',
				address:'تهران سهروردی',
				rate:'4.7',
				likes:'225000',
				description:"اگر در منطقه سهروردی بدنبال یک قنادی معروف باشید که طرفداران خود را از سایر نقاط تهران به آنجا میکشاند، آن قنادی ناتلی خواهد بود. این مجموعه فعالیت خود را از سال 1340 شروع کرده است. در طبقه دوم این مجموعه کافه وجود دارد که میتوانید در کنار کیک های فوق العاده آن از گذرندان وقت در آنجا لذت ببرید. افرادی که سالهاست بعنوان مشتریان دائمی به این قنادی می آیند همواره خوردن پای شاتوت را به شما توصیه می کنند. همچنین مافین، کاپ کیک، دونات و کیک شکلاتی نیز یکی از پر طرفدارترین محصولات است که هر سلیقه ای را راضی نگه میدارد."
			},
				{
					id:2,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D8%B4%DB%8C%D8%B1%DB%8C%D9%86%DB%8C-%D8%A8%DB%8C-%D8%A8%DB%8C-600x334.jpg',
					title:'شیرینی بی بی\n',
					averagePrice:'24000',
					address:'تهران یوسف آباد. میدان کلانتری خیابان مهرام',
					rate:'4.7',
					likes:'225000',
					description:"کیفیت بالای محصولات در کنار سابقه بالای قنادی این اطمینان را به شما میدهد که با خیالی آسوده و بدون مقایسه به خرید بپردازید. شیرینی بی بی که ابتدا کار خود را از یوسف آباد شروع کرد، در پخت شیرینی خشک به معروف بود اما به مرور کیک های زیبا و شیرینی خامه ای با خامه های سبک توانست در بین مشتریان به محبوبیت برسد.\n" +
						"\n" +
						"اگر به تاریخچه تاسیس آن رجوع کنید خواهید دید در سال 59 در محله یوسف آباد قنادی به نام “شیرینی سرا” تاسیس شد که بعدها به بی بی تغییر نام داد. کیک تولد های این قنادی هم یکی از محبوبترین هاست. یک نوع باقلوا در این قنادی تهیه میشود که بسیار لذیذ است و بین طرفداران شکم دهان به دهان می چرخد."
				},
				{
					id:3,
					image:'https://images.unsplash.com/photo-1562548210-9107ddb7004a?ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60',
					title:'شیرینی نیشکر',
					averagePrice:'24000',
					address:' تهران فرمانیه بلوار اندرزگو روبروی بانک پاسارگاد',
					rate:'4.7',
					likes:'225000',
					description:"این قنادی هم یکی از برندهای این صنعت است. در وبسایت رسمی این قنادی می خوانیم که دارای لوح تقدیر از وزارت بهداشت و درمان بوده و دارای ایزو مدیریت کیفیت 9001-2008 می باشد. همچنین با داشتن 5 شعبه مستقل، پیشرو در ارائه خدمات به وزارت خانه ها و سمینارها و مناسبت های مختلف است.\n" +
						"\n" +
						"شعبه مرکزی این شیرینی فروشی در فرمانیه قرار دارد و شعبه VIP در زعفرانیه دایر است. بعد از سالها فعالیت هنوز هم محصولاتی مانند چیز کیک آن بسیار معروف است. تارت های میوه ای هم یکی دیگر از محصولات پر طرفدار آنهاست."
				},
				{
					id:4,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D8%B4%DB%8C%D8%B1%DB%8C%D9%86%DB%8C-%D8%B3%D9%88%D9%84%D8%AF%D9%88%D8%B4-600x398.jpg',
					title:'شیرینی سولدوش\n',
					averagePrice:'24000',
					address:'پاسداران- میدان هروی',
					rate:'4.7',
					likes:'225000',
					description:"در منطقه پاسداران قنادی معروفی هست که با یکبار مراجعه به آن تبدیل به مشتریان همیشگی آن میشوید. جدیدترین دسرهای فرانسوی و تارت ها را در این قنادی خواهید یافت. همچنین کیک های عروسی این قنادی از شهرت بالایی برخوردار است. اگر به فکر کیک چند طبقه هستید، از این فروشگاه دیدن کنید. در بین بهترین شیرینی فروشی های تهران سولدوش هم جایگاه خاصی دارد.\n" +
						"\n" +
						"سال 1345 آقای قادر نوروزی سنگ بنای این قنادی را گذاشت و این سیر رشد تا اکنون ادامه دارد تا حدی که لوح تقدیر از وزارت بهداشت و درمان هم دریافت کرده است. اگر در ماه رمضان از جلوی این فروشگاه گذر کنید خواهید دید افراد زیادی در انتظار زولبیا و بامیه های معروف این قنادی هستند."
				},
				{
					id:5,
					image:'https://images.unsplash.com/photo-1542141372-98a047557466?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60',
					title:'شیرینی دانمارکی\n',
					averagePrice:'24000',
					address:'تهران خیابان ویلا- بالاتر از طالقانی',
					rate:'4.7',
					likes:'225000',
					description:"در خیابان ویلا قدیمی ترین قنادی به اسم دانمارکی وجود دارد که نمیتوان بوی شیرینی های داغ دانمارکی مخصوصا در فصل سرد سال به سادگی گذشت. این قنادی از ساعت 8:30 کار خود را شروع می کند و شاهد خواهید بود از همان ساعات اولیه مشتریان به آن رجوع می کنند.\n" +
						"\n" +
						"اگر علاقمند به خوردن دانمارکی تازه و داغ هستید در ساعت 10 صبح و 4 بعد از ظهر آنجا باشید تا با طعم دانمارکی شکری که از محبوبترین محصولات آن است کام خود ار شیرین کنید. شعبه دوم این قنادی نیز در پاسداران با همان موفقیت و محبوبیت در حال فعالیت است."
				},
				{
					id:6,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D8%B4%DB%8C%D8%B1%DB%8C%D9%86%DB%8C-%D9%84%D8%A7%D8%AF%D9%86-600x240.jpg',
					title:'قنادی لادن\n',
					averagePrice:'24000',
					address:'تهران ولنجک بلوار دانشجو',
					rate:'4.7',
					likes:'225000',
					description: "قنادی لادن ولنجک نیز خاطرات زیبایی را برای اهالی رقم زده است. بیش از نیم قرن اعتبار و فعالیت مناسب در این صنعت آنرا به برندی قابل تحسین تبدیل کرده است. کیک های تزئینی و آجیل و خشکبار مخصوص خود از محصولاتی است که این قنادی به خود می بالد. در ماههای اخیر نیز با افزودن قابلیت سفارش اینترنتی شیرینی گامی دیگر در جهت خدمت به مشتریان برداشته است."
				},
				{
					id:7,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D8%AE%D8%A7%D9%86%D9%87-%DA%A9%DB%8C%DA%A9-%D9%BE%D9%84%D8%A7%D8%AA%DB%8C%D9%86-600x400.jpg',
					title:'کافه قنادی لرد\n',
					averagePrice:'24000',
					address:'تهران – خیابان استاد نجات الهی – نرسیده به کریمخان زند',
					rate:'4.7',
					likes:'225000',
					description:"اگر در منطقه کریمخان هستید راهی ندارید جز انتخاب یک قنادی که در خیابان ویلا قرار دارد و آن نامی نیست جز قنادی لرد. این قنادی از سال 1344 پذیرای میهمانی ها و جشن های ما بوده و تارت های آن از محبوبیت بالایی برخوردارند. مشتریان زیادی را در این شیرینی فروشی خواهید دید که علاوه بر کیک و شیرینی عالی از فضای کافه آن نیز بخوبی استفاده می کنند.\n" +
						"\n" +
						"قهوه های ترک در این قنادی هم از شهرت بالایی برخوردار است و میتوانید آنرا با چیز کیک یا شیرینی خامه ای میل کنید. هنگامی ه وارد این قنادی میشوید بوی قهوه و وانیل سرمستتان می کند و بیاد دوران کودکی می افتید که شیرینی های خامه ای بزرگ هوش از سر میبرد."
				},
				{
					id:8,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D9%82%D9%86%D8%A7%D8%AF%DB%8C-%D8%A2%D9%82-%D8%A8%D8%A7%D9%86%D9%88.jpeg',
					title:'قنادی آق بانو\n',
					averagePrice:'24000',
					address:'تهران زیر پل حافظ جنب تالار وحدت',
					rate:'4.7',
					likes:'225000',
					description:"اگر در مرکز شهر سکونت دارید حتما سری به قنادی های اطراف پارک دانشجو بزنید. ما شیرینی آق بانو را به شما پیشنهاد می کنیم. آجیل های تازه در کنار شیرینی خشک با کیفیت آنرا به میزبانی مورد اعتماد برای میهمانی های ما بدل کرده است."
				},
				{
					id:9,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D8%B4%DB%8C%D8%B1%DB%8C%D9%86%DB%8C-%D9%81%D8%B1%D9%88%D8%B4%DB%8C-%DA%A9%D9%88%DA%A9.jpg',
					title:'شیرینی کوک',
					averagePrice:'24000',
					address:' تهران خیابان میرداماد ساختمان اسکان',
					rate:'4.7',
					likes:'225000',
					description:"اگر بدنبال یک اثر هنری در خوراکی ها هستید حتما لیست منو های قنادی کوک را ببینید. کیک پزان خوش ذوق این قنادی از طرح های بینظیر و خلاقیت خود برای تزئین استفاده می کنند. شعبه های مختلف این شیرینی فروشی در میرداماد، قیطریه، ولنجک، هروی، سعادت آباد و شهرک غرب نشان از مدیریت خوب و محبوبیت بالای این برند است."
				},
				{
					id:10,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D9%82%D9%86%D8%A7%D8%AF%DB%8C-%D9%87%D8%A7%D9%86%D8%B3-600x400.jpg',
					title:'شیرینی هانس\n',
					averagePrice:'24000',
					address:'Tehran Azadi Shahidan St',
					rate:'4.7',
					likes:'225000',
					description:"شیرینی هانس نیز یکی از قدیمی ترین و معروفترین شیرینی فروشی های تهران است که صاحبان آن ایرانی نبودند. چیزی که آنرا نسبت به بقیه قنادی ها متمایز می کند ظاهر ساده و با صفای آن است که بیشتر شبیه خانه است تا قنادی و سراسر آنرا گیاهان و گلها پوشانده است. مشتریان آن فقط برای خوردن کیک اسفنجی لایه لایه حاضرند از هرجای تهران خود را به آنجا برسانند. اگر قصد دارید سری به این قنادی بزنید باید از قبل تماس بگیرید و یا مدت زیادی را در صف منتظر باشید."
				},
				{
					id:11,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D9%82%D9%86%D8%A7%D8%AF%DB%8C-%D8%B5%D8%A7%D8%AD%D8%A8%D9%82%D8%B1%D8%A7%D9%86%DB%8C%D9%87-600x375.jpg',
					title:'قنادی صاحبقرانیه\n',
					averagePrice:'24000',
					address:'تهران سه راه اقدسیه نرسیده به پارک نیاوران',
					rate:'4.7',
					likes:'225000',
					description: "اگر بدنبال کیک های با شکوه برای عروسی هستید و قیمت آن خیلی برای شما مهم نیست ما قنادی صاحبقرانیه را به شما توصیه می کنیم. از ابتدای ورود خود محو دیزاین و ظاهر چشم نواز این قنادی میشوید. در کیک های مختلف از نظرات شما برای پر کردن لایه های مختلف هم استفاده میشود و در آخر از خمیر فوندانت برای روکش کیک استفاده می کنند تا ماندگاری آنرا افزایش دهند.\n" +
						"\n" +
						"کیک های سه بعدی و اسفنجی نیز از شهرت بالایی برخوردارند و در کل برای میهمانی های مجلل میتوانید روی این شیرینی فروشی حساب کنید."
				},
				{
					id:12,
					image:'https://shiriniyab.com/wp-content/uploads/2018/08/%D8%B4%DB%8C%D8%B1%DB%8C%D9%86%DB%8C-%D9%81%D8%B1%D8%A7%D9%86%D8%B3%D9%87-600x466.jpg',
					title:'شیرینی فرانسه\n',
					averagePrice:'24000',
					address:'تهران سه تقاطع آزادی وصال',
					rate:'4.7',
					likes:'225000',
					description: "بسیاری از دانشجویان و کتاب خوان های خیابان انقلاب یک پاتوق خوب دارند که حتما یکبار به آن مراجعه کرده اند. 53 سال است که یک قنادی در این منطقه بهترین خدمات را ارائه میدهد و گاتا های خوشمزه از محبوبترین محصولاتش است.\n" +
						"\n" +
						"همچنین در روزهای سرد زمستان میتوانید از طعم شیر کاکائوهای آن لذت ببرید. دانشجویان این مکان را خیلی دوست دارند زیرا چند نسل از این قنادی خاطرات خوب و دلنشین دارند."
				}
			]
		)
			.then(result=>{
				return result
			})
	}
};
