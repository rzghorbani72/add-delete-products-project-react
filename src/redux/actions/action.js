import {GET_POSTS, RESET_POSTS, UPDATE_POSTS} from './type';
import {get} from '../../static/defaultPosts';

export function getPosts (type) {
	switch (type) {
		case GET_POSTS://Get fetch remain items
			return get()
				.then(data=>{
				return {
					type: GET_POSTS,
					payload: data
				};
			});
		default://RESET fetch all items deleted+remain
			return get()
				.then(data => {
					return {
						type: RESET_POSTS,
						payload: data
					}
				})
	}
};
export function updatePosts (id) {
	return get()
		.then(data => {
			return {
				type: UPDATE_POSTS,
				payload: {
					id: id,
					info: data
				}
			}
		});
};

export const getPostsAction = data => dispatch =>{
	return getPosts(data).then(action=>dispatch(action));
};
export const updatePostsAction = data => dispatch =>{
	return updatePosts(data).then(action=>dispatch(action));
};
