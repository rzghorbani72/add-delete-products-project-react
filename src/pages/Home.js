import React from 'react';
import {connect} from 'react-redux';
import {isEmpty} from 'lodash';
import {getPostsAction, updatePostsAction} from "../redux/actions/action";
import {GET_POSTS} from "../redux/actions/type";
import Header from "../components/header";

export class Home extends React.Component {
	componentDidMount() {
		this.props.getPostsAction(GET_POSTS);
	}
	render() {
		const {postsInfo} = this.props;
		const posts = isEmpty(postsInfo) ? [] : postsInfo.posts;
		return (
			<React.Fragment>
				<Header/>
				<div id="Index" className="col-12 row pageWrapper">
					<div className={`resetBtn`}>
						<button type='button' onClick={() => this.props.getPostsAction()} className="alert alert-success">بارگیری مجدد آیتم ها</button>
					</div>
					<div className={`container col-12 row`}>
						{isEmpty(posts) ? '' :
							posts.map((info, i) => (
								<div key={i} className={`col-lg-3 col-md-4 col-sm-6 col-xs-12 cardWrapper`}>
									<div className={`card mx-auto`}>
										<img src={info.image} alt={info.title}/>
										<div className={`itemName`}><a href={`/post/${info.id}`}>{info.title}</a></div>
										<div className={`col-12 row`}>
											<div className={`col-lg-6 col-sm-12 alert alert-danger actionBtn deleteBtn`}>
												<button type='button' className={`alert-danger delete-${info.id}`} onClick={() => this.props.updatePostsAction(info.id)}>حذف</button>
											</div>
											<div className={`col-lg-6 col-sm-12 alert alert-info actionBtn`}>
												<a href={`/post/${info.id}`}>
													<button  type='button' className={`alert-info`}>مشاهده جزییات</button>
												</a>
											</div>
										</div>
									</div>
								</div>
							))
						}
					</div>
				</div>
			</React.Fragment>
		);
	}
}

const mapStateToProps = state => ({
	...state,
	output: state.output
});
export default connect(mapStateToProps, {
	getPostsAction,
	updatePostsAction
})(Home);
