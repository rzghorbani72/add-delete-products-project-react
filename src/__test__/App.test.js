import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Home from '../pages/Home';
import {BrowserRouter, BrowserRouter as Router} from "react-router-dom";
import {get} from "../static/defaultPosts";
import {GET_POSTS, RESET_POSTS, UPDATE_POSTS} from '../redux/actions/type'
import {getPosts, updatePosts} from '../redux/actions/action'
import reducer from '../redux/reducers';
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import _ from "lodash";
import About from "../pages/About";
import Post from "../pages/Post";
import Header from "../components/header";

Enzyme.configure({adapter: new Adapter()})

function setup(store, page) {
	const props = initialProps;
	const enzymeWrapper = mount(<BrowserRouter>
		<Provider store={store}>
			<Router>
				{page === 'Home' ? <Home {...initialProps}/> :
					page === 'Post' ? <Post {...initialPostProps} /> :
						page === 'About' ? <About/> : <Header/>
				}
			</Router>
		</Provider>
	</BrowserRouter>);
	return {
		props,
		enzymeWrapper
	}
}
function componentRenderer (store,page){
	const component = renderer.create(
		<BrowserRouter>
			<Provider store={store}>
				<Router>
					{page === 'Home' ? <Home {...initialProps}/> :
						page === 'Post' ? <Post {...initialPostProps} /> :
							page === 'About' ? <About/> : <Header/>
					}
				</Router>
			</Provider>
		</BrowserRouter>
	);
	return component;
};
const middlewares = [thunk];
const initialState = {
	errors: {},
	postsInfo: {
		deleted: [],
		posts: []
	},
}
const initialPostProps = {
	match : {
		params:{
			id:1
		}
	}
};
const initialProps = {
	getPosts: jest.fn()
};
const mockStore = configureStore(middlewares);
describe('App Test', () => {
	let store,
		component,
		sampleMockJsonArray = [];
	beforeAll(() => {
		return get()
			.then(data => {
				sampleMockJsonArray = data;
			})
	});

	
	describe('App React-Redux Test', () => {
		
		describe('App snapshots', () => {
				store = mockStore({
					errors: {},
					postsInfo: {posts: sampleMockJsonArray},
					deleted: []
				});
				store.dispatch = jest.fn();
			it('should render HomePage with given state from Redux store', () => {
				component = componentRenderer(store,'Home')
				expect(component.toJSON())
					.toMatchSnapshot();
				expect(store.dispatch.mock.calls)
					.toMatchSnapshot()
			});
			it('should render PostPage with given state from Redux store', () => {
				component = componentRenderer(store,'Post');
				expect(component.toJSON())
					.toMatchSnapshot();
				expect(store.dispatch.mock.calls)
					.toMatchSnapshot()
			});
			it('should render AboutPage with given state from Redux store', () => {
				component = componentRenderer(store,'About');
				expect(component.toJSON())
					.toMatchSnapshot();
				expect(store.dispatch.mock.calls)
					.toMatchSnapshot()
			});
		});
		describe('Actions', () => {
			it('getPosts_Action should fetch [GET_POSTS] dispatch payload', () => {
				const expectedAction = {
					type: GET_POSTS,
					payload: sampleMockJsonArray
				};
				return getPosts(GET_POSTS)
					.then(data => {
						expect(data)
							.toEqual(expectedAction)
					})
			});
			it('getPosts_Action should fetch [RESET_POSTS] dispatch payload', () => {
				const expectedAction = {
					type: RESET_POSTS,
					payload: sampleMockJsonArray
				};
				return getPosts()
					.then(data => {
						expect(data)
							.toEqual(expectedAction)
					})
			});
			it('updatePosts_Action should delete one item and verify dispatch payload', () => {
				const expected = {
					type: UPDATE_POSTS,
					payload: {
						id: 5,
						info: sampleMockJsonArray
					}
				};
				return updatePosts(5)
					.then(data => {
						expect(data)
							.toEqual(expected)
					})
			})
		});
		describe('Reducers', () => {
			it('should return the initial state', () => {
				expect(reducer(undefined, {}))
					.toEqual(initialState)
			});
			it('should handle postReducer', () => {
				expect(
					reducer(initialState, {
						type: GET_POSTS,
						payload: sampleMockJsonArray
					})
				)
					.toEqual(
						{
							errors: {},
							postsInfo: {
								deleted: [],
								posts: sampleMockJsonArray
							}
						}
					);
				
				expect(
					reducer(
						initialState,
						{
							type: RESET_POSTS,
							payload: sampleMockJsonArray
						}
					)
				)
					.toEqual(
						{
							errors: {},
							postsInfo: {
								deleted: [],
								posts: sampleMockJsonArray
							}
						}
					);
				
				expect(
					reducer(
						initialState,
						{
							type: UPDATE_POSTS,
							payload: {
								id: 5,
								info: sampleMockJsonArray
							}
						}
					)
				)
					.toEqual(
						{
							errors: {},
							postsInfo: {
								deleted: [5],
								posts: filteredPosts(sampleMockJsonArray, [5])
							}
						}
					);
				
				function filteredPosts(items, deletedIds) {
					_.remove(items, (item) => {return _.includes(deletedIds, item.id)});
					return items;
				}
			})
		});
	});
	describe('App Components Test', () => {
		describe('App Page self components', () => {
			it('home page self render', () => {
				const {enzymeWrapper} = setup(store,'Home');
				expect(enzymeWrapper.find('.alert-success')
					.text())
					.toEqual('بارگیری مجدد آیتم ها');
			});
			it('header self render', () => {
				const {enzymeWrapper} = setup(store);
				expect(enzymeWrapper.find('#HomeLink')
					.first()
					.text())
					.toEqual("خانه");
				expect(enzymeWrapper.find('#AboutLink')
					.first()
					.text())
					.toEqual("درباره‌ما");
			});
			it('About page self render', () => {
				const {enzymeWrapper} = setup(store,'About');
				expect(enzymeWrapper.find('h2').first()
					.text())
					.toEqual('درباره ما');
			});
		})
	});
});
