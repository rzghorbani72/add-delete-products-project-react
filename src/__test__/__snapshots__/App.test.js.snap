// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test App React-Redux Test App snapshots should render AboutPage with given state from Redux store 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="nav"
    >
      <a
        className="nav-item nav-link active "
        href="/"
        id="HomeLink"
        onClick={[Function]}
      >
        خانه
      </a>
      <a
        className="nav-item nav-link "
        href="/about"
        id="AboutLink"
        onClick={[Function]}
      >
        درباره‌ما
      </a>
    </div>
  </header>,
  <div
    className="pageWrapper col-12 row"
    id="About"
  >
    <div
      className="col-lg-3 col-sm-12"
    />
    <div
      className="col-lg-6 col-sm-12 cardWrapper"
    >
      <div>
        <h2>
          درباره ما
        </h2>
      </div>
      <div
        className="desctiption"
      >
        <span>
          لورم ایپسوم یا طرح‌نما (به انگلیسی: Lorem ipsum) به متنی آزمایشی و بی‌معنی در صنعت چاپ، صفحه‌آرایی و طراحی گرافیک گفته می‌شود. طراح گرافیک از این متن به عنوان عنصری از ترکیب بندی برای پر کردن صفحه و ارایه اولیه شکل ظاهری و کلی طرح سفارش گرفته شده استفاده می نماید، تا از نظر گرافیکی نشانگر چگونگی نوع و اندازه فونت و ظاهر متن باشد. معمولا طراحان گرافیک برای صفحه‌آرایی، نخست از متن‌های آزمایشی و بی‌معنی استفاده می‌کنند تا صرفا به مشتری یا صاحب کار خود نشان دهند که صفحه طراحی یا صفحه بندی شده بعد از اینکه متن در آن قرار گیرد چگونه به نظر می‌رسد و قلم‌ها و اندازه‌بندی‌ها چگونه در نظر گرفته شده‌است. از آنجایی که طراحان عموما نویسنده متن نیستند و وظیفه رعایت حق تکثیر متون را ندارند و در همان حال کار آنها به نوعی وابسته به متن می‌باشد آنها با استفاده از محتویات ساختگی، صفحه گرافیکی خود را صفحه‌آرایی می‌کنند تا مرحله طراحی و صفحه‌بندی را به پایان برند. توسط فرم زیر می توانید متن ساختگی مورد نظر خود را در واحدهای کاراکتر، کلمه و پاراگراف تولید کنید، سپس آنرا کپی کنید و در کار مورد نظر خود قرار دهید. زبان مورد نظر خود را انتخاب کنید ...
        </span>
      </div>
    </div>
    <div
      className="col-lg-3 col-sm-12"
    />
  </div>,
]
`;

exports[`App Test App React-Redux Test App snapshots should render AboutPage with given state from Redux store 2`] = `
Array [
  Array [
    [Function],
  ],
]
`;

exports[`App Test App React-Redux Test App snapshots should render HomePage with given state from Redux store 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="nav"
    >
      <a
        className="nav-item nav-link active "
        href="/"
        id="HomeLink"
        onClick={[Function]}
      >
        خانه
      </a>
      <a
        className="nav-item nav-link "
        href="/about"
        id="AboutLink"
        onClick={[Function]}
      >
        درباره‌ما
      </a>
    </div>
  </header>,
  <div
    className="col-12 row pageWrapper"
    id="Index"
  >
    <div
      className="resetBtn"
    >
      <button
        className="alert alert-success"
        onClick={[Function]}
        type="button"
      >
        بارگیری مجدد آیتم ها
      </button>
    </div>
    <div
      className="container col-12 row"
    >
      
    </div>
  </div>,
]
`;

exports[`App Test App React-Redux Test App snapshots should render HomePage with given state from Redux store 2`] = `
Array [
  Array [
    [Function],
  ],
]
`;

exports[`App Test App React-Redux Test App snapshots should render PostPage with given state from Redux store 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="nav"
    >
      <a
        className="nav-item nav-link active "
        href="/"
        id="HomeLink"
        onClick={[Function]}
      >
        خانه
      </a>
      <a
        className="nav-item nav-link "
        href="/about"
        id="AboutLink"
        onClick={[Function]}
      >
        درباره‌ما
      </a>
    </div>
  </header>,
  <div
    className="pageWrapper col-12 row"
    id="Post"
  >
    <div
      className="col-lg-3 col-sm-12"
    />
    
    <div
      className="col-lg-3 col-sm-12"
    />
  </div>,
]
`;

exports[`App Test App React-Redux Test App snapshots should render PostPage with given state from Redux store 2`] = `
Array [
  Array [
    [Function],
  ],
]
`;
